src/dmqproto/client/request/internal/Consume.d

```diff
@@ -378,574 +152,596 @@ public struct Consume
-                {
-                    payload.add(Consume.cmd.code);
-                    payload.add(Consume.cmd.ver);
-                    payload.addArray(this.context.user_params.args.channel);
-                    payload.addArray(this.context.user_params.args.subscriber);
-                    payload.add(start_state);
-                }
-            );
+        ***********************************************************************/
 
-            // Receive status from node and stop the request if not Ok
-            auto status = conn.receiveValue!(StatusCode)();
-            if ( !Consume.handleGlobalStatusCodes(status, context,
-                conn.remote_address) )
-            {
-                switch ( status )
-                {
-                    case RequestStatusCode.Started:
-                        break;
+        private bool stopped = false;
 
-                    case RequestStatusCode.Error:
-                        // The node returned an error code. Notify the user and
-                        // end the request on this connection.
-                        Notification n;
-                        n.node_error = NodeInfo(this.conn.remote_address);
-                        Consume.notify(this.context.user_params, n);
-                        return State.Exit;
-
-                default:
-                    log.warn("Received unknown status code {} from node in "
-                        ~ "response to Consume request. Treating as Error.",
-                        status);
-                    goto case RequestStatusCode.Error;
-                }
-            }
+        /// Token passed to fiber suspend/resume calls.
+        private static MessageFiber.Token token =
+            MessageFiber.Token(typeof(this).stringof);
+
+        /// Constructor, starts the fiber.
+        private this ( )
+        {
+            this.batch_buffer = cast(Const!(void)[]*)this.outer.resources.getBuffer();
+            this.fiber = this.outer.resources.getFiber(&this.fiberMethod);
+            this.fiber.start();
         }
-        finally
+
+        /***********************************************************************
+
+            Adds a batch of records to be passed to the user notifier and
+            resumes the fiber if it is waiting for more records. Called by the
+            `Reader` when a `Records` message from the node has arrived.
+
+            Params:
+                record_batch = the batch of records to add
+
+        ***********************************************************************/
+
+        public void addRecords ( in void[] record_batch )
         {
-            assert(this.context.shared_working.handlers_waiting_for_ack);
-            --this.context.shared_working.handlers_waiting_for_ack;
+            // Append record_batch to *this.batch_buffer, which may or may not
+            // be empty.
+
+            if (!(*this.batch_buffer).length)
+                enableStomping(*this.batch_buffer);
```

Comment:
When compiling wave in D2, I get an error on this line:
ocean/transition.d(205): Error: static assert  "Must not call `enableStomping` on const/immutable array"
dmqproto/client/request/internal/Consume.d(423):        instantiated from here: enableStomping!(const(void))

Reply:
Oh dear. Thanks for reporting.